modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useEffect} from 'react';
+
+/**
+ * Hook for excuting a function callback when clicking outside
+ * some DOM elements (for example, a ClayPopover)
+ */
+export default function useOnClickOutside(
+	elements: Array<any>,
+	handler: (...args: any[]) => any
+) {
```

Comment:
I think we can narrow this types a bit here: 

* Elements: since we are calling `closest` and `contains` functions we can use the interface [Node](https://developer.mozilla.org/en-US/docs/Web/API/Node/contains) here as `Array<Node>`
* handler: handler is only going to receive an argument, element, and we don't expect it to return anything, so we can do `handler: (event: Event) => void`

What do you think? :)

Reply:
Thank you for the feedback. I'm going narrow the types a little bit :)